pkg/kube/controllers/queue.go

```diff
@@ -157,8 +157,9 @@ func (q Queue) processNextItem() bool {
 
 	err := q.workFn(key)
 	if err != nil {
-		if q.queue.NumRequeues(key) < q.maxAttempts {
-			q.log.Errorf("error handling %v, retrying: %v", key, err)
+		retryCount := q.queue.NumRequeues(key)
```

Comment:
Does numRequeues get reset after we successfully process it? We might have > maxAttempts updates to the same key.

Reply:
Looks like it will behave how we want.